sound/soc/sof/intel/hda-dsp.c

```diff
@@ -669,6 +675,9 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 	/* power down all hda links */
 	hda_bus_ml_suspend(bus);
 
+	if (sdev->dspless_mode_selected)
```

Comment:
do we need these checks in the suspend/resum callbacks even though we will never invoke these in case of dspless mode?

Reply:
These functions will be used, not sure what makes you think otherwise?

the tests  protect the power_down stuff, which should not happen, and access to PP bar registers.

Reply:
Oh i got confused with this line:
https://github.com/thesofproject/linux/blob/9b5c7765a81bbd778e2c7b7a32dbe681c00b51d8/sound/soc/sof/pm.c#L106

Its checked after the resume op is invoked.

Reply:
We should be able to turn things off and back on on suspend/resume even if the DSP is not used.

Reply:
not getting the comment @ujfalusi what are you trying to say? If there is no DSP we cannot turn it off nor disable the capabilities (PP_BAR not enabled).

Reply:
@plbossart, even if we don't have DSP we still need to do something on suspend/resume with HDA alone.

Reply:
that's right @plbossart but that's what this PR is doing. We only skip the DSP parts, not sure what is wrong here?

Reply:
@plbossart I dont think anything is wrong here. I think @ujfalusi was answering to my question

Reply:
no change then